src/Builder.hs

```diff
@@ -19,10 +19,13 @@ import Stage
 -- 1) Compiling sources into object files.
 -- 2) Extracting source dependencies, e.g. by passing -M command line argument.
 -- 3) Linking object files & static libraries into an executable.
-data CompilerMode = Compile
-                  | FindDependencies
-                  | Link
-                  deriving (Eq, Generic, Show)
+-- We have CcMode for CC and GhcMode for GHC.
+
+data CcMode = CcCompile | CcFindDependencies
+    deriving (Eq, Generic, Show)
+
+data GhcMode = GhcCompile | GhcFindDependencies | GhcLink
+    deriving (Eq, Generic, Show)
 
```

Comment:
I think I have a better idea for naming modes:
```haskell
data CcMode = CompileC | FindCDependencies

data GhcMode = CompileHs | FindHsDependencies | LinkHs
```
We'll later add `CompileCmm` and `CompileS` to the latter.